pkgs/applications/science/chemistry/cp2k/default.nix

```diff
@@ -1,15 +1,59 @@
-{ lib, stdenv, fetchFromGitHub, mpiCheckPhaseHook, python3, gfortran, blas, lapack
-, fftw, libint, libvori, libxc, mpi, gsl, scalapack, openssh, makeWrapper
-, libxsmm, spglib, which, pkg-config, plumed, zlib
+{ lib
+, stdenv
+, fetchFromGitHub
+, mpiCheckPhaseHook
+, python3
+, gfortran
+, blas
+, lapack
+, fftw
+, libint
+, libvori
+, libxc
+, mpi
+, gsl
+, scalapack
+, openssh
+, makeWrapper
+, libxsmm
+, spglib
+, which
+, pkg-config
+, plumed
+, zlib
+, hdf5-fortran
+, sirius
+, libvdwxc
+, spla
+, spfft
 , enableElpa ? false
 , elpa
-} :
+, gpuBackend ? "none"
+, cudaPackages
+, gpuVersion ? "Mi100"
+, gpuArch ? "gfx908"
```

Comment:
This seem very specific. Is there a reason for this combo? Does it apply to rocm, CUDA, or both?

Reply:
It is, yes. `gpuVersion` applies to both Cuda and ROCM, `gpuArch` only to ROCM. The reason for this specific combination is merely the CP2K documentation, where this is the example and it is stated that this is the combination on which the ROCM things were developed and tested. Others are possible and specific to what ROCM supports.

For CUDA, at least the gpuVersion would need to be changed to, e.g. `P100`. The CP2K docs have an incomplete list.

Reply:
Then maybe add comment in the code here.